### ISSUE TITLE
add callback that saves full model/checkpoints

### ISSUE THREAD

**Issue by philippmuench** [hiddengenome/deepG#55]
- currently only when the last epoch finished, the model will be saved https://github.com/hiddengenome/deepG/blob/master/R/train.R#L217-L225 therefore progess get lost when its not finished
- "checkpoints" e.g. model wights or full model should be written to disk periodically, usually this can be done using a callback https://keras.rstudio.com/reference/callback_model_checkpoint.html
- nice would be if file name of the checkpoint includes the the loss or accuracy in the file name, that one can easy identify which was the best e.g. `runname1_checkpoint2_0.9821.hdf5`
- think ok to keep all checkpoints produced after every epoch, not overwrite the checkpoint
- if easy implementable, highlight the best checkpoint/model, e.g. if checkpoint is better move it to a different folder and replace the old "best" model
- inform the user where checkpoints will be written using `message()`. also allow to change path to checkpoint folder as function argument
- check if the checkpoint saves the file in the same format as here

``` 
 Rmodel <-
    keras::serialize_model(model, include_optimizer = TRUE)
  save(Rmodel, file = paste0(run.name, "_full_model.Rdata"))
  keras::save_model_hdf5(
    model,
    paste0(run.name, "_full_model.hdf5"),
    overwrite = TRUE,
    include_optimizer = TRUE
  )
```

otherwise, maybe add a custom callback that saves the checkpoint as "serialized"
